pkg/apiserver/registry/security/workloadidentity/storage/tokenrequest.go

```diff
@@ -0,0 +1,158 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package storage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/gardener/gardener/pkg/api"
+	securityapi "github.com/gardener/gardener/pkg/apis/security"
+	securityv1alpha1 "github.com/gardener/gardener/pkg/apis/security/v1alpha1"
+	securityvalidation "github.com/gardener/gardener/pkg/apis/security/validation"
+)
+
+// TokenRequestREST implements a RESTStorage for a token request.
+type TokenRequestREST struct {
+	// client kubernetes.Interface // Might be needed later to read the context object
+
+	minDuration            int64
+	maxDuration            int64
+	workloadIdentityGetter getter
+}
+
+type getter interface {
+	Get(context.Context, string, *metav1.GetOptions) (runtime.Object, error)
+}
+
+var (
+	_   = rest.NamedCreater(&TokenRequestREST{})
+	_   = rest.GroupVersionKindProvider(&TokenRequestREST{})
+	gvk = securityv1alpha1.SchemeGroupVersion.WithKind("TokenRequest")
+)
+
+// New returns an instance of the object.
+func (r *TokenRequestREST) New() runtime.Object {
+	return &securityv1alpha1.TokenRequest{}
+}
+
+// Destroy cleans up its resources on shutdown.
+func (r *TokenRequestREST) Destroy() {
+	// Given that underlying store is shared with REST, we don't destroy it here explicitly.
+}
+
+// Create returns a TokenRequest with workload identity token based on
+// - spec of the workload identity
+// - spec of the token request
+// - referenced context object
+// - gardener installation
+func (r *TokenRequestREST) Create(ctx context.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, _ *metav1.CreateOptions) (runtime.Object, error) {
+	if createValidation != nil {
+		if err := createValidation(ctx, obj.DeepCopyObject()); err != nil {
+			return nil, err
+		}
+	}
+
+	tokenRequest := &securityapi.TokenRequest{}
+	if err := api.Scheme.Convert(obj, tokenRequest, nil); err != nil {
+		return nil, fmt.Errorf("failed converting %T to %T: %w", obj, tokenRequest, err)
+	}
+	if errs := securityvalidation.ValidateTokenRequest(tokenRequest); len(errs) != 0 {
+		return nil, apierrors.NewInvalid(gvk.GroupKind(), "", errs)
+	}
+
+	// TODO(vpnachev): implement context specific features
+	// if tokenRequest.Spec.ContextObject != nil {	}
+
+	wiObj, err := r.workloadIdentityGetter.Get(ctx, name, &metav1.GetOptions{})
```

Comment:
Maybe it's only me, but I'd prefer to use `workloadIdentity` instead of `wi`. It is a bit longer, but it improves readability of the code.